src/features/todos/ToDoSlice.js

```diff
@@ -0,0 +1,26 @@
+import { createSlice } from '@reduxjs/toolkit';
+
+let id = 1
+
+export const ToDoSlice = createSlice({
+    name: "todos",
+    initialState: [],
+    reducers: {
+        addToDo: {
+            reducer: (state, action) => { state.push(action.payload)},
+            prepare: (todo) => ({payload: {id: id++, text:todo ,completed: false}})
+        },
+        toggleToDo: (state, action) => 
+            state.map((todo) => todo.id === action.payload
+            if(todo) {
```

Comment:
you want to have the `if` logic `if(todo.id === action.payload)` then update that specific todo